src/main/java/org/example/marketplacebackend/DTO/outgoing/WatchListResponseDTO.java

```diff
@@ -0,0 +1,8 @@
+package org.example.marketplacebackend.DTO.outgoing;
+
+import java.util.UUID;
+import org.example.marketplacebackend.model.ProductCategory;
+
+public record WatchListResponseDTO(UUID id, ProductCategory productCategory) {
```

Comment:
DTOs shouldn't contain model classes. Swap the ProductCategory model class for your ProductCategoryDTO and this should be good to go.